libdevice/sanitizer_utils.cpp

```diff
@@ -251,6 +257,12 @@ bool __asan_region_is_value(uptr addr, int32_t as, std::size_t size,
   return true;
 }
 
+#if __SYCL_DEVICE_ONLY__
+#define __DEVICE_SANITIZER_REPORT_ACCESSOR __DeviceSanitizerReportMem.get()
+#else
+#define __DEVICE_SANITIZER_REPORT_ACCESSOR
```

Comment:
Could you please elaborate a bit on what the expected outcome in this `#else` branch is? Looks like the compilation will just fail in that case.

Reply:
Hi, @sergey-semenov 
Yes, the file must be built with __SYCL_DEVICE_ONLY__ or corresponding openmp macro defined, it not, we expect a compfail to be reported for notification. The openmp support will be added later.
Thanks very much